lib/stale.js

```diff
@@ -54,9 +56,11 @@ module.exports = class Stale {
       this.logger.trace({ owner, repo }, 'Configured to close stale issues')
       const closableItems = (await this.getClosable(type)).data.items
 
-      await Promise.all(closableItems.filter(issue => !issue.locked).map(issue => {
-        this.close(type, issue)
-      }))
+      await Promise.all(
+        closableItems
+          .filter(issue => !issue.locked && issue.state !== 'closed')
+          .map(issue => this.close(type, issue))
```

Comment:
Note that this subtle change to actually return the result of the call to `this.close` fixes another potential race condition like the one I addressed in #270.